creusot/tests/should_succeed/inc_some_tree.rs

```diff
@@ -0,0 +1,66 @@
+#![feature(register_tool, rustc_attrs)]
+#![register_tool(creusot)]
+#![feature(proc_macro_hygiene, stmt_expr_attributes)]
+
+extern crate creusot_contracts;
+use creusot_contracts::*;
+
+enum Tree {
+  Node(Box<Tree>, u32, Box<Tree>),
+  Leaf,
+}
+use Tree::*;
+
+logic! {
+  fn sum_tree(t: Tree) -> Int {
+  match t {
+    Node(tl, a, tr) => sum_tree(*tl) + Int::from(a) + sum_tree(*tr),
+    Leaf => 0,
+  }
+  }
+}
+
+/* TODO: prove this lemma */
+#[trusted]
+#[ensures(sum_tree(*t) >= 0)]
+fn lemma_sum_tree_nonneg(t: &Tree) {}
```

Comment:
i think you should be able to prove these trees by doing induction in the function body. If you do 

```rust
fn lemma_sum_tree_nonneg(t: &Tree) {
match t {
Leaf => (),
Node(box l, n, box r) =>{ lemma_sum_tree_nonneg(l); lemma_sum_tree_nonneg(r) }
}
```

Reply:
Ah, I see.

Reply:
The problem is that this makes `lemma_sum_tree_nonneg` actually non-trivial as Rust code, taking unboundedly many program steps.
It is nice if we can mark lemma functions like `lemma_sum_tree_nonneg` as ghost code and eliminate them in the actual executable.

Reply:
Is this going to be supported in Creusot?

Reply:
>  It is nice if we can mark lemma functions like lemma_sum_tree_nonneg as ghost code and eliminate them in the actual executable.

Agreed. We (Claude, JH and me) discussed this this morning. I have a clear plan on how to add 'logic functions with contracts' which is what you need.

Reply:
The solution is actually what's outlined in #28, but we only use the logical part.

Reply:
I'd say give it 1-2 weeks for addition but if you write your "lemma"s in this style you would only need to add a `#[pure]` attribute to be able to use them in a _true_ ghost context.

Reply:
OK, then I'll write lemmas like that and just comment out the `#[pure]` annotations for now.